### PR TITLE
rpc: create buffered subscriptions on /subscribe

### LINKED ISSUE #3935: RPC subscriptions: client is not pulling messages fast enough
**Tendermint version**
0.32.3

**ABCI app**
- kvstore
- InProc implementation

**Environment**:
- **OS** linux-5.2.10 (Archlinux)
- **Install tools**: git

**What happened**:
When I subscribe to events via RPC API, after getting 0 or more events (this is completely random) I get the following error:
```json
{
  "jsonrpc": "2.0",
  "id": "1#event",
  "error": {
    "code": -32000,
    "message": "Server error",
    "data": "subscription was cancelled (reason: client is not pulling messages fast enough)"
  }
}
```

**What you expected to happen**:
I should keep getting the events

**Have you tried the latest version**: yes

**How to reproduce it** (as minimally and precisely as possible):
- step 1
Start tendermint:
`go run ./cmd/tendermint/main.go node --proxy_app kvstore`

- step 2
Subscribe to events, I'm using [ws](https://github.com/yhat/ws) but any WS client should be sufficient:
```bash
$  ws ws://localhost:26657/websocket
> {"jsonrpc":"2.0", "id": 1, "method": "subscribe", "params": {"query": "tm.event='Tx'"}}
< {
  "jsonrpc": "2.0",
  "id": 1,
  "result": {}
}

- step 3
Broadcast tons of Tx:
```bash
WAIT=0.1
for i in $(seq 10000)
do
  echo $i
  sleep $WAIT
  curl http://localhost:26657/broadcast_tx_sync\?tx\=\"$i\"
done
```

You can increment `$WAIT` to 1 or more seconds until the error dissapears.

**Config**:
Generated by `tendermint init`

### PR DESCRIPTION
**Pull request by melekes**
buffer size: 100

Closes #3935

______

For contributor use:

- [ ] ~~Wrote tests~~ tested manually
- [x] Updated CHANGELOG_PENDING.md
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Updated relevant documentation (`docs/`) and code comments
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: rpc/core/events.go [hunk 1]
```diff
@@ -12,6 +12,11 @@ import (
 	rpctypes "github.com/tendermint/tendermint/rpc/lib/types"
 )
 
+const (
+	// Buffer on the Tendermint (server) side to allow some slowness in clients.
+	subCap = 100
```
Comment (erikgrinaker) [inline rpc/core/events.go]: This name is a bit unclear. Maybe `subBufferSize`?

File: CHANGELOG_PENDING.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ Friendly reminder, we have a [bug bounty program](https://hackerone.com/tendermi
 
 ### BUG FIXES:
 
+- [rpc] \#3935 Create buffered subscriptions on `/subscribe` (@melekes)
```
Comment (tessr) [inline CHANGELOG_PENDING.md]: nit, but do you want to make this a link like the other items in this list?

Comment (marbar3778) [reply CHANGELOG_PENDING.md]: technically this is the correct way according to the contributing.md and then there is a script to add all the links prior to release. People have been putting links recently, I guess we should decide on one way or the other?

Comment (melekes) [reply CHANGELOG_PENDING.md]: yeah, there's a script for that

**Pull request merged**